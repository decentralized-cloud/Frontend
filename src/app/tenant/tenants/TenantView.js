import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';

import { tenantProp } from './PropTypes';
import Styles from './Styles';

const TenantView = ({ tenant: { name } }) => {
  const classes = Styles();

  return (
    <TableRow>
      <TableCell padding="checkbox">
        <Checkbox checked={false} />
      </TableCell>
      <TableCell component="th" scope="row" padding="none">
        <Link className={classes.link}>{name}</Link>
      </TableCell>
    </TableRow>
  );
};

TenantView.propTypes = {
  tenant: tenantProp.isRequired,
};

export default TenantView;
