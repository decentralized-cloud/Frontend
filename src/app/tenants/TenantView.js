import React from 'react';

const TenantView = ({ tenant }) => <h1>{tenant.name}</h1>;

export default TenantView;
