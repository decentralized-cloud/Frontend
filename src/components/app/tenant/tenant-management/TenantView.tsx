import React from 'react';
import graphql from 'babel-plugin-relay/macro';
import { createFragmentContainer } from 'react-relay';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';

import { TenantView_tenant } from './__generated__/TenantView_tenant.graphql';
import Styles from './Styles';

interface TenantViewProps {
  tenant: TenantView_tenant;
  onTenantClick: (id: string) => void;
}

export const TenantView = React.memo<TenantViewProps>(({ tenant: { id, name }, onTenantClick }) => {
  const classes = Styles();

  return (
    <TableRow>
      <TableCell padding="checkbox">
        <Checkbox checked={false} />
      </TableCell>
      <TableCell component="th" scope="row" padding="none">
        <Link className={classes.link} onClick={() => onTenantClick(id)}>
          {name}
        </Link>
      </TableCell>
    </TableRow>
  );
});

export default createFragmentContainer(TenantView, {
  tenant: graphql`
    fragment TenantView_tenant on Tenant {
      id
      name
    }
  `,
});
