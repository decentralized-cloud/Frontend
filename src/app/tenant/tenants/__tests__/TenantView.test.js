import React from 'react';
import renderer from 'react-test-renderer';

import TenantView from '../TenantView';

describe('<TenantView />', () => {
  const tenant = {
    id: 'tenantID',
    name: 'Tenant Name',
  };

  const props = {
    onTenantClick: jest.fn(),
  };

  it('renders correctly', () => {
    const tree = renderer.create(<TenantView tenant={tenant} {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
