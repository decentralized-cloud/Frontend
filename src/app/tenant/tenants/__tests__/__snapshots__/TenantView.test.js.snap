// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenantView /> renders correctly 1`] = `
<tr
  className="MuiTableRow-root"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-paddingCheckbox"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-34"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </td>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-paddingNone"
    scope="row"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-7 MuiTypography-colorPrimary"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
    >
      Tenant Name
    </a>
  </th>
</tr>
`;
